pkg/scalers/azure_servicebus_scaler.go

```diff
@@ -233,105 +237,100 @@ func (s *azureServiceBusScaler) GetMetrics(ctx context.Context, metricName strin
-}
-
 // Returns the length of the queue or subscription
 func (s *azureServiceBusScaler) getAzureServiceBusLength(ctx context.Context) (int64, error) {
-	// get namespace
-	namespace, err := s.getServiceBusNamespace(ctx)
+	// get adminClient
+	adminClient, err := s.getServiceBusAdminClient(ctx)
 	if err != nil {
 		return -1, err
 	}
 	// switch case for queue vs topic here
 	switch s.metadata.entityType {
 	case queue:
-		return getQueueEntityFromNamespace(ctx, namespace, s.metadata.queueName)
+		return getQueueLength(ctx, adminClient, s.metadata.queueName)
 	case subscription:
-		return getSubscriptionEntityFromNamespace(ctx, namespace, s.metadata.topicName, s.metadata.subscriptionName)
+		return getSubscriptionLength(ctx, adminClient, s.metadata.topicName, s.metadata.subscriptionName)
 	default:
 		return -1, fmt.Errorf("no entity type")
 	}
 }
 
 // Returns service bus namespace object
-func (s *azureServiceBusScaler) getServiceBusNamespace(ctx context.Context) (*servicebus.Namespace, error) {
-	var namespace *servicebus.Namespace
+func (s *azureServiceBusScaler) getServiceBusAdminClient(ctx context.Context) (*admin.Client, error) {
+	if s.client != nil {
+		return s.client, nil
+	}
+
+	var adminClient *admin.Client
 	var err error
 
 	switch s.podIdentity.Provider {
 	case "", kedav1alpha1.PodIdentityProviderNone:
-		namespace, err = servicebus.NewNamespace(servicebus.NamespaceWithConnectionString(s.metadata.connection))
+		adminClient, err = admin.NewClientFromConnectionString(s.metadata.connection, nil)
 		if err != nil {
-			return namespace, err
+			return nil, err
 		}
 	case kedav1alpha1.PodIdentityProviderAzure, kedav1alpha1.PodIdentityProviderAzureWorkload:
-		namespace, err = servicebus.NewNamespace()
+		// Once azure-sdk-for-go supports Workload Identity we can remove this and use default implementation
+		// https://github.com/Azure/azure-sdk-for-go/issues/15615
+		var creds []azcore.TokenCredential
+		options := &azidentity.DefaultAzureCredentialOptions{}
+
+		envCred, err := azidentity.NewEnvironmentCredential(&azidentity.EnvironmentCredentialOptions{ClientOptions: options.ClientOptions})
+		if err == nil {
+			creds = append(creds, envCred)
+		}
+
+		cliCred, err := azidentity.NewAzureCLICredential(&azidentity.AzureCLICredentialOptions{TenantID: options.TenantID})
+		if err == nil {
+			creds = append(creds, cliCred)
+		}
+
+		wiCred := azure.NewADWorkloadIdentityCredential(ctx, s.podIdentity.IdentityID, serviceBusResource)
+		creds = append(creds, wiCred)
+
+		managedIdentityOptions := &azidentity.ManagedIdentityCredentialOptions{ClientOptions: options.ClientOptions}
+		if ID, ok := os.LookupEnv("AZURE_CLIENT_ID"); ok {
+			managedIdentityOptions.ID = azidentity.ClientID(ID)
+		}
+		msiCred, err := azidentity.NewManagedIdentityCredential(managedIdentityOptions)
+		if err == nil {
+			creds = append(creds, msiCred)
+		}
+
+		chain, err := azidentity.NewChainedTokenCredential(creds, nil)
```

Comment:
I am slightly confused here, why do we need to use all of these creds here?

Reply:
And if we do require all of these creds, would it not be better to separate them into two cases `PodIdentityProviderAzure` and `PodIdentityProviderAzureWorkload`. In the case of Azure we can use the `NewDefaultCredentials` and in case of Azure Workload our custom credential struct.

Reply:
In fact, can't we just do this 

1. Case Pod Identity - Use `ManagedIdentityCredentials`
1. Case Workload Identity - Use `ADWorkloadIdentityCredentials`

Reply:
In fact yes, we could just set those 2 cases and that's it but, knowing that they expect to introduce `ADWorkloadIdentityCredentials` into `NewDefaultCredentials`  in next version (v1.2.0), I prefer to have only one method for both because once they support it, we can easily replace this code to a single `NewDefaultCredentials`.

We can split it for the moment in 2 cases, but I'd like to have `NewAzureCLICredential` in both for local debugging.
WDYT if we do something like:
1. Case Pod Identity - Use `NewDefaultCredentials`
2. Case Workload Identity - Use `NewChainedTokenCredential` with `NewAzureCLICredential` and `ADWorkloadIdentityCredentials`

Once the SDK full support WI, we can merge them into a single `NewDefaultCredentials` if any pod identity is used.
If we merge both in one single case, we could eventually merge both identity providers in a single one, simplifying the scenario having only 1 pod identity for azure who supports both

Reply:
I have already split in 2 cases, managed identities and workload identities, both with support to AzureCLI for debugging and local development. 
I have reviewed the internal implementation and it doesn't try with all the auth mechanism all the time. It just iterates them once to select the proper credential provider and then caches it for the future, so supporting AzCLI is a good way to improve the local development I think.

Reply:
The good part is that based on their milestone, we can expect it in one month approx 😃